### ISSUE TITLE
Push to GitHub gh-pages branch fails during Travis CI build

### ISSUE THREAD

**Issue by atsakiridis** [travis-ci/travis-ci#6900]
Hello,

As part of my build I generate appledoc documentation and try to push it to GitHub gh-pages. The problem is that the push fails with:
```
$ git push git@github.com:RestComm/restcomm-ios-sdk.git gh-pages
ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
Host key verification failed.
fatal: Could not read from remote repository.
```
Some notes:
- I have verified that my encrypted key is properly decrypted in the Travis CI env, by hashing it and checking with my actual private key file (followed https://docs.travis-ci.com/user/encrypting-files/ to set this up)
- I have added my public key to the GitHub repo so that the push is accepted and have tested it from my local env
- I have added the private key to the ssh agent with ssh-add
- I have ensured that my private key is not password protected
- I have made sure that the push is made towards the ssh GitHub endpoint, not https

Here's an [example build](https://travis-ci.org/RestComm/restcomm-ios-sdk/builds/177048094) in case you want to have a look (notice that the build is successful for some reason, yet the push hasn't been made)

Best regards,
Antonis

**Comment by BanzaiMan** [comment]
Looks like your GitHub authentication is not properly set up. It is set to ask for your ssh password, it appears to me. Even if `ssh-askpass` were available, your push will probably fail because the build script can't supply it (and GitHub doesn't allow ssh connexion to github.com).

To automate this step, you can use [git credentials helper](https://git-scm.com/docs/gitcredentials), along with a special-purpose GitHub OAuth token.

Create a new token at https://github.com/settings/tokens with proper authorization (I believe "repo:public_repo" is sufficient, but maybe you'd need "repo" instead), export it (say, `$GITHUB_OAUTH_TOKEN`; remember to either encrypt it or use the settings page), then:

```yaml
before_install:
  - git config credential.helper "store --file=.git/credentials"; echo "https://${GITHUB_OAUTH_TOKEN}:@github.com" > .git/credentials 2>/dev/null
  - git config --global user.email "YOU@EXAMPLE.COM"
  - git config --global user.name "YOUR_NAME"
```

**Comment by atsakiridis** [comment]
Thanks a lot for your prompt reply @BanzaiMan, I will try the Credentials Helper and post back.

Meanwhile, on the second part of your answer, I thought that using a GitHub [Deploy Key](https://developer.github.com/guides/managing-deploy-keys/#deploy-keys) and adding its private counterpart  in Travis CI with `ssh-add`(after having it decrypted, to avoid private key exposure of course), would be enough to be able to push to that repo using the ssh endpoint `git@github.com:RestComm/restcomm-ios-sdk.git`. Isn't that so?

In fact I tested that locally by creating a brand new user in my OSX computer (to make sure I would be starting fresh, without any configuration) and it seems to be working fine. 

If that is the case why should I also create a [Personal Access Token](https://github.com/settings/tokens)? Am I missing something here?

**Comment by atsakiridis** [comment]
@BanzaiMan, I tested what you proposed and and works great, so thanks a lot! Is this solution documented somewhere? Cause from my search on the internet I was directed the Deploy Key way which as discussed I failed to properly utilize it for some reason (still curious, any idea why that is?). If not, I 'd be happy to contribute a blog about this to hopefully help others who try the same.

I'm closing as it fixed my issue
